Add Select{Next|Previous|Nearest} macros

binff04 opened the pull request:
If you play without Razor on macOS or Linux, this is something that you probably miss a lot.

SelectNext: iterates through selected type of mobiles in a 12-tile radius (from top to bottom of the list);
SelectPrevious: iterates through selected type of mobiles in a 12-tile radius (from bottom to top of the list);
SelectNearest: iterates through selected type of mobiles in a 12-tile radius (from top to bottom of the list but only among the nearest mobiles);

binff04 committed:
Add Select{Next|Previous|Nearest} macros

If you play without Razor on macOS or Linux, this is something that you probably miss a lot.

SelectNext: iterates through selected type of mobiles in a 12-tile radius (from top to bottom of the list);
SelectPrevious: iterates through selected type of mobiles in a 12-tile radius (from bottom to top of the list);
SelectNearest: iterates through selected type of mobiles in a 12-tile radius (from top to bottom of the list but only among the nearest mobiles);

andreakarasho commented on src/Game/Managers/TargetManager.cs:
```diff
@@ -283,5 +283,176 @@ public static void SendMultiTarget(ushort x, ushort y, sbyte z)
         }
+
+        public static bool IsMobileSelectableAsTarget(Mobile mobile, int type)
+        {
```
It’s a good practice adding a null check for the mobile var or pass a Serial as argument and search the mobile with World.Mobiles dictionary then add a null check.
This is to avoid Ghost reference and unexpected crahes if we want call this method out of here for any reason.
I prefer the second option honestly.

binff04 replied:
I totally agree. I made the changes, so both methods now expect Serial instead of Mobile, get mobile from World.Mobiles and check for null. No place for mistake.

andreakarasho commented on src/Game/Managers/TargetManager.cs:
```diff
@@ -283,5 +283,176 @@ public static void SendMultiTarget(ushort x, ushort y, sbyte z)
+
+        public static void SetLastTarget(Mobile target)
+        {
+            GameActions.MessageOverhead($"Target: {target.Name}", Notoriety.GetHue(target.NotorietyFlag), World.Player);
```
The same as above

binff04 replied:
Yeah, same change.

binff04 committed:
Improve null checking

Also, instead of passing Mobile objects to `IsMobileSelectableAsTarget` and `SetLastTarget` methods, pass Serial and get mobile from `World.Mobiles` list.
